lib/Frontend/ModuleInterfaceSupport.cpp

```diff
@@ -317,6 +322,17 @@ class InheritedProtocolCollector {
       return;
 
     map[nominal].recordProtocols(directlyInherited, D);
+    // Collect protocols inherited from super classes
```

Comment:
I still think it would be better to look at nominal->getLocalConformances(), and filter out any InheritedProtocolConformances from the resulting list.